lib/basedriver/helpers.js

```diff
@@ -96,37 +97,44 @@ async function configureApp (app, supportedAppExtensions) {
       }
 
       let fileName = null;
+      const basename = sanitize(path.basename(decodeURIComponent(pathname)), {
+        replacement: SANITIZE_REPLACEMENT
+      });
+      const extname = path.extname(basename);
       // to determine if we need to unzip the app, we have a number of places
       // to look: content type, content disposition, or the file extension
-      if (ZIP_EXTS.includes(path.extname(newApp))) {
-        fileName = sanitize(path.basename(newApp), {
-          replacement: SANITIZE_REPLACEMENT
-        });
+      if (ZIP_EXTS.includes(extname)) {
+        fileName = basename;
         shouldUnzipApp = true;
       }
       if (headers['content-type']) {
         logger.debug(`Content-Type: ${headers['content-type']}`);
         // the filetype may not be obvious for certain urls, so check the mime type too
         if (ZIP_MIME_TYPES.includes(headers['content-type'])) {
           if (!fileName) {
-            fileName = 'appium-app.zip';
+            fileName = `${DEFAULT_BASENAME}.zip`;
           }
           shouldUnzipApp = true;
         }
       }
       if (headers['content-disposition'] && /^attachment/i.test(headers['content-disposition'])) {
+        logger.debug(`Content-Disposition: ${headers['content-disposition']}`);
         const match = /filename="([^"]+)/i.exec(headers['content-disposition']);
         if (match) {
-          logger.debug(`Parsed file name '${match[1]}' from 'Content-Disposition' header`);
           fileName = sanitize(match[1], {
             replacement: SANITIZE_REPLACEMENT
           });
           shouldUnzipApp = shouldUnzipApp || ZIP_EXTS.includes(path.extname(fileName));
         }
       }
       if (!fileName) {
-        // assign the default extension if none has been detected
-        fileName = `appium-app${_.first(supportedAppExtensions)}`;
+        // assign the default file name and the extension if none has been detected
+        fileName = basename
+          ? basename.substring(0, basename.length - extname.length)
+          : DEFAULT_BASENAME;
+        fileName += supportedAppExtensions.includes(extname)
```

Comment:
Does this mean that if the `extname` is not supported it will just default to the first supported extension?

If so, maybe we should put a log message saying something like `'${extName}' is not supported. Defaulting to ${_.first(supportedAppExtensions}`